modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/portlet/action/GetContentDashboardItemInfoMVCResourceCommand.java

```diff
@@ -138,6 +138,8 @@ protected void doServeResource(
 					_toString(contentDashboardItem.getCreateDate())
 				).put(
 					"description", contentDashboardItem.getDescription(locale)
+				).put(
+					"downloadURL", contentDashboardItem.getDownloadURL()
```

Comment:
Hey @claraizquierdo !!

We don't add URLs to the API, we have already a download Action, could you take a look to the method _getFetchSharingButtonURL and obtain the Download URL the same way?

In the end you will need to mimic the _getDownloadURL method in the FilesEntryContentDashboardItem class:

```
private String _getDownloadURL() {
		return Optional.ofNullable(
			ServiceContextThreadLocal.getServiceContext()
		).map(
			ServiceContext::getLiferayPortletRequest
		).map(
			portletRequest -> {
				List<ContentDashboardItemAction> contentDashboardItemActions =
					getContentDashboardItemActions(
						_portal.getHttpServletRequest(portletRequest),
						ContentDashboardItemAction.Type.DOWNLOAD);

				if (!contentDashboardItemActions.isEmpty()) {
					ContentDashboardItemAction contentDashboardItemAction =
						contentDashboardItemActions.get(0);

					return contentDashboardItemAction.getURL();
				}

				return null;
			}
		).orElse(
			null
		);
	}
```

Thanks!!

Reply:
I can do it myself, but as you start to do the java code, if you want to continue, fell free.

Also, it is usually not  no mix changes in different modules.

We will usually create a commit for the changes in the API, another commit for the baseline, another commit for all the implementation, another commit with the changes in the content-dashboard-web.

Thanks!!

Reply:
Hi @cgoncas , thank you for the review! I will try to do it as you say :)